persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java

```diff
@@ -107,160 +115,181 @@ public void start() throws PersistenceException {
       } else {
          marshaller = ctx.getPersistenceMarshaller();
       }
-      ConfigurationBuilder builder = buildRemoteConfiguration(configuration, marshaller);
-      remoteCacheManager = new RemoteCacheManager(builder.build());
 
-      if (configuration.remoteCacheName().isEmpty())
-         remoteCache = remoteCacheManager.getCache();
-      else
-         remoteCache = remoteCacheManager.getCache(configuration.remoteCacheName());
+      // We currently don't support segmentation when not shared as we will have to support remove segments -
+      // should we add this?
+      if (needsStateTransfer && !configuration.shared() && configuration.segmented()) {
```

Comment:
Since ISPN-10749, should we not also throw an exception for invalidation mode?

Reply:
Yes, I was actually in the middle of rewriting all of this and found different issues. To be honest I am not going to be able to fix all the problems with the RemoteStore in time for the 11.0.1 build, so I am closing this PR for now.